packages/babel-generator/src/printer.js

```diff
@@ -5,12 +5,258 @@ import Buffer from "./buffer";
 import * as n from "./node";
 import * as t from "babel-types";
 
-export default class Printer extends Buffer {
-  constructor(...args) {
-    super(...args);
+export default class Printer {
+  constructor(format, map) {
+    this._format = format || {};
+    this._buf = new Buffer(map);
     this.insideAux = false;
     this.printAuxAfterOnNextUserNode = false;
     this._printStack = [];
+    this.printedCommentStarts = {};
+    this.parenPushNewlineState = null;
+    this._indent = 0;
+  }
+
+  printedCommentStarts: Object;
+  parenPushNewlineState: ?Object;
+
+  /**
+   * Get the current indent.
+   */
+
+  _getIndent(): string {
+    if (this._format.compact || this._format.concise) {
+      return "";
+    } else {
+      return repeat(this._format.indent.style, this._indent);
+    }
+  }
+
+  /**
+   * Increment indent size.
+   */
+
+  indent() {
+    this._indent++;
+  }
+
+  /**
+   * Decrement indent size.
+   */
+
+  dedent() {
+    this._indent--;
+  }
+
+  /**
+   * Add a semicolon to the buffer.
+   */
+
+  semicolon() {
+    this._maybeIndent();
+    this._buf.semicolon();
+    this._endsWithWord = false;
+  }
+
+  /**
+   * Add a right brace to the buffer.
+   */
+
+  rightBrace() {
+    if (!this.endsWith("\n")) this.newline();
+
+    if (this._format.minified && !this._lastPrintedIsEmptyStatement) {
+      this._buf.removeLastSemicolon();
+    }
+    this.token("}");
+  }
+
+  /**
+   * Add a keyword to the buffer.
+   */
+
+  keyword(name: string) {
+    this.word(name);
+    this.space();
+  }
+
+  /**
+   * Add a space to the buffer unless it is compact.
+   */
+
+  space(force: boolean = false) {
+    if (this._format.compact) return;
+
+    if ((this._buf.hasContent() && !this.endsWith(" ") && !this.endsWith("\n")) || force) {
```

Comment:
I left it this way because the `force` param here is only `true` when printing `VariableDeclaration` nodes, so on the whole it's probably going to be `false` the vast majority of the time.